File: algorithms/java/BubbleSort.java
import java.util.*;

class BubbleSort {
    public static void main(String[] args) {
        int arr[] = {4, 12, 23, 1, 90, 1243, 98, 24, 1,56, 124};
        bubbleSort(arr);
        System.out.println("Array after sorting is: ");
        for(int i : arr) {
            System.out.print(i + " ");
        }
    }

    public static void bubbleSort(int arr[]) {
        int n = arr.length;
        boolean swapped;
        for (int i = 0; i < n - 1; i++) {
            swapped = false;
            for (int j = 0; j < n - i - 1; j++) { 
                if (arr[j] > arr[j + 1]) { 
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                    swapped = true;
                }
            }
            if (swapped == false)
                break;
        }
    }
}

File: algorithms/python/bubble_sort.py
def bubbleSort(arr):
    n = len(arr)
    for i in range(n):
        swapped = False
        for j in range(0, n-i-1):
            if(arr[j+1] < arr[j]):
                arr[j+1],arr[j] = arr[j], arr[j+1]
                swapped = True
        if swapped == False:
            break;

arr = [23, 112, 2, 124, 56, 90, 124, 78, 41, 98]
bubbleSort(arr)
print("Sorted array is: ")
for i in range(len(arr)):
    print("%d" %arr[i], end=" ")